lib/state-controller.js

```diff
@@ -76,66 +69,60 @@ const StateController = {
   downloadKite(url, opts) {
     opts = opts || {};
     return this.canInstallKite()
-    .then(() => utils.spawnPromise('curl', [
-      '-L', url, '--output', this.KITE_DMG_PATH,
-    ], 'curl_error'))
+    .then(() => this.streamKiteDownload(opts.onDownloadProgress))
     .then(() => utils.guardCall(opts.onDownload))
     .then(() => opts.install && this.installKite(opts));
   },
 
-  installKite(opts) {
-    opts = opts || {};
+  streamKiteDownload(progress) {
+    const req = https.request(this.releaseURL);
+    req.end();
+
+    return utils.promisifyRequest(req)
+    .then(resp => {
+      if (resp.statusCode === 303) {
```

Comment:
Actually it's the implementation prior to the switch to curl, but it make totally sense to guard against that case.